k-NN Add ADC and Random Rotation documentation to binary quantization (#10657)
opensearch-project/documentation-website: _vector-search/optimizing-storage/binary-quantization.md

```
{% include copy-curl.html %}

> ## Enhancing search quality with ADC and RR

Reviewer: ```suggestion
## Enhancing search quality with ADC and RR
**Introduced 3.2**
{: .label .label-purple }
```

> If the recall is lacking, you can specify asymmetric distance computation (ADC) or random rotation (RR) in the index mapping to apply enhanced search capabilities. These techniques are available in OpenSearch 3.2 and later versions.

Reviewer: ```suggestion
If your search results have low recall, you can apply enhanced search capabilities by specifying asymmetric distance computation (ADC) or random rotation (RR) in the index mapping.
```

Author: How about "If your search results have low recall, you can enhance search quality by specifying asymmetric distance computation (ADC) or random rotation (RR) in the index mapping."?

ADC maintains a full-precision query vector while rescaling it to have meaningful distance computations against binary-quantized document vectors. This asymmetric approach preserves more information about the query vector, boosting search quality without significant memory penalty. ADC is supported for 1-bit quantization only.

> Random rotation addresses the issue where binary quantization gives equal weight to each vector dimension during the quantization process. By rotating the distribution, RR can "smooth" variance (information) from high-variance dimensions into low-variance dimensions, preserving more information during the 32x compression process. RR is supported for 1-bit, 2-bit, and 4-bit quantization.

Reviewer: ```suggestion
Random rotation addresses the issue where binary quantization gives equal weight to each vector dimension during the quantization process. By rotating the distribution, RR can balance variance (information) from high-variance dimensions into low-variance dimensions, preserving more information during the 32x compression process. RR is supported for 1-bit, 2-bit, and 4-bit quantization.
```

Reviewer: Maybe "balance", "redistribute", or "reallocate" instead of "smooth"?

Author: Good point, I think "redistribute" is most accurate

For optimal performance and recall enhancement, use both ADC and RR together:

```json
PUT vector-index
{
  "settings" : {
    "index": {
      "knn": true
    }
  },
  "mappings": {
    "properties": {
      "vector_field": {
        "type": "knn_vector",
        "dimension": 8,
        "method": {
            "name": "hnsw",
            "engine": "faiss",
            "space_type": "l2",
            "parameters": {
              "encoder": {
                "name": "binary",
                "parameters": {
                  "bits": 1,
                  "random_rotation": true,
                  "enable_adc": true
                }
              }
            }
        }
      }
    }
  }
}
```
{% include copy-curl.html %}

> **Note:** ADC and RR impact search or indexing performance, so they are opt-in features. ADC may introduce a moderate latency increase due to full-precision distance computations, while RR primarily affects indexing latency as vectors must be rotated during the process.

Reviewer: ```suggestion
ADC and RR impact search or indexing performance, so they are disabled by default. ADC may introduce a moderate latency increase because of full-precision distance computations, while RR primarily affects indexing latency because vectors must be rotated during the process.
{: .note}
```
